src/lib/utils.ts

```diff
@@ -259,14 +260,10 @@ export function encodeKey(
         const dataElement = value[index];
         if (index === 0) {
           // This is arrayLength as the first element in the raw array
+          // encoded as a uint128
           results.push({
             key: schema.key,
-            value: encodeKeyValue(
-              'Number',
-              'uint256',
-              value.length.toString(),
-              schema.name,
-            ) as string,
+            value: leftPad(value.length, 32),
```

Comment:
What is the reason for not using the `encodeKeyValue`?

`leftPad(value.length, 32)` is duplicated code with https://github.com/ERC725Alliance/erc725.js/pull/287/files#diff-5927840c3323c66acd20762ba79459d6e5fe967c32d6d2a4be8751fb589c0c93R330

We want only one place in the code that does the encoding, not to sprint it everywhere in the code no?

Reply:
I think because it was encoding as a valueContent "Number" and the "Number" valueContent in the code uses uint256 as the valueType to encode.

Reply:
@Hugoo here, after debugging with @kalote we found that `encodeKeyValue` ends up encoding the value as a `Number`, which itself encodes as a `uint256` (and we want to encode Array length as `uint128`.

https://github.com/ERC725Alliance/erc725.js/blob/9e075c4318e0d0055b52b3b70c4ad50ac06c1412/src/lib/encoder.ts#L413-L428

There is a better way, using `encodeValueType` function (already used in this file).